sky/tools/create_macos_gen_snapshots.py

```diff
@@ -11,28 +11,45 @@
 
 
 def main():
-  parser = argparse.ArgumentParser(description='Copies architecture-dependent gen_snapshot binaries to output dir')
+  parser = argparse.ArgumentParser(
+      description='Copies architecture-dependent gen_snapshot binaries to output dir'
+  )
 
   parser.add_argument('--dst', type=str, required=True)
+  parser.add_argument('--x64-out-dir', type=str)
   parser.add_argument('--arm64-out-dir', type=str)
   parser.add_argument('--armv7-out-dir', type=str)
 
   args = parser.parse_args()
 
+  if args.x64_out_dir:
+    generate_gen_snapshot(
+        args.x64_out_dir,
+        os.path.join(args.dst, 'gen_snapshot_x64')
+    )
+
   if args.arm64_out_dir:
-    generate_gen_snapshot(args.arm64_out_dir, os.path.join(args.dst, 'gen_snapshot_arm64'))
+    generate_gen_snapshot(
+        os.path.join(args.arm64_out_dir, 'clang_x64'),
+        os.path.join(args.dst, 'gen_snapshot_arm64')
+    )
 
   if args.armv7_out_dir:
-    generate_gen_snapshot(args.armv7_out_dir, os.path.join(args.dst, 'gen_snapshot_armv7'))
+    generate_gen_snapshot(
```

Comment:
Why are we doing building anything `armv7` for macOS?

Reply:
This is also the script that's used to package the gen_snapshot binaries for iOS. Instead of adding a macOS specific script, I'm just adding an x64 case to the existing script.

Reply:
Gotcha, I knew you didn't change anything with that, I was just surprised to see it in `create_macos_gen_snapshots`

Reply:
may related to #10430